scripts/Jenkinsfiles/quality

```diff
@@ -0,0 +1,219 @@
+def runQualityTests() {
+    sshagent(credentials: ['jenkins-worker'], ignoreMissing: true) {
+        // Determine git refspec, branch, and clone type
+        git_shallow_clone = true
+        git_extensions = []
+        if (env.ghprbActualCommit) {
+            git_branch = "${ghprbActualCommit}"
+            refspec = "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*"
+            if (SHARD == "4") {
+                git_shallow_clone = false
+                git_extensions.add([$class: 'WipeWorkspace'])
```

Comment:
Do we want to wipe all workspaces to make sure that nothing from a previous build is present?

Reply:
Hm, good question. If so, it should happen in the initial Jenkinsfile checkout, which means a change to my DSL PR.

Reply:
I don't think this is actually a concern, since each clone exists in a workspace specific to each build. So even if a worker is re-used there shouldn't be any issue. The main reason why I've added it in here is because each worker automatically does a shallow clone to fetch the jenkinsfile, and I need to replace it with a more detailed checkout under certain cirumstances.

Reply:
Got it, thanks for explaining